src/engine/runtime.js

```diff
@@ -19,14 +19,21 @@ var defaultBlockPackages = {
 
 /**
  * Manages targets, scripts, and the sequencer.
+ * @param {!RenderWebGL} renderer Renderer for the VM
  */
-function Runtime () {
+function Runtime (renderer) {
     // Bind event emitter
     EventEmitter.call(this);
 
     // State for the runtime
 
     /**
+     * Renderer
+     * @type {!RenderWebGL}
+     */
+    this.renderer = renderer;
```

Comment:
Having it a property of runtime seems fine to me. But I also think it should be optional/possible to be missing.